ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/search_tree/algorithms/SearchTree.java

```diff
@@ -118,6 +139,8 @@ public CompletableFuture<OptimizationResult> run(SearchTreeInput searchTreeInput
         this.prePerimeterRangeActionSetPoints = new HashMap<>();
         rootLeaf.getRangeActions().stream().forEach(rangeAction -> prePerimeterRangeActionSetPoints.put(rangeAction, prePerimeterOutput.getOptimizedSetPoint(rangeAction)));
 
+        applyForcedNetworkActionsOnRootLeaf();
```

Comment:
maybe we could remove the forced network actions from the set of available network actions too to avoid adding and removing them (since they were already applied) as combinations in the bloomer

Reply:
the bloomer does automatically remove them, thanks to the re-construction of the root leaf (check cucumber test 16.4.3)

Reply:
yes ok